obspy/signal/src/aic_simple.c

```diff
@@ -0,0 +1,59 @@
+/**
+ * Simple AIC (Akaike information criterion) by Maeda (1985).
+ * Author: Danylo Ulianych
+ * Copyright (C) 2022 D. Ulianych
+ */
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <math.h>
+
+
+typedef struct {
+	double mean;
+	double varsum;  // variance sum
+	uint32_t count;
+} OnlineMean;
+
+
+void OnlineMean_Init(OnlineMean *oMean, double firstVal) {
+    oMean->mean = firstVal;
+    oMean->varsum = 0;
+    oMean->count = 1;
+}
+
+/**
+ * Welford's online algorithm of estimation the population mean & variance.
+ */
+void OnlineMean_Update(OnlineMean *oMean, double newVal) {
+    oMean->count++;
+    double delta = newVal - oMean->mean;
+    oMean->mean += delta / oMean->count;
+    oMean->varsum += delta * (newVal - oMean->mean);
+}
+
+
+void aic_simple(double *aic, const double *arr, uint32_t size) {
+    OnlineMean oMean;
+
+    // special case: the forward loop skips the first element; init with zero
+    aic[0] = 0;
+
+    OnlineMean_Init(&oMean, arr[0]);
+    for (uint32_t i = 1; i < size - 1; i++) {
+        // forward loop: AIC(i) = (i + 1) log(Var{0..i})
+        OnlineMean_Update(&oMean, arr[i]);
+        aic[i] = oMean.count * log(oMean.varsum / oMean.count);
+    }
+
+    OnlineMean_Init(&oMean, arr[size - 1]);
+    for (uint32_t i = size - 2; i > 0; i--) {
```

Comment:
And `i` will wrap if `size == 1` (I realize you've checked in Python, but still double check here if anyone bypasses that.)

Reply:
I'm sure I saw C implementations for some other methods in the `trigger.py` without the double-checking in the C code.
Anyway.
You haven't specified how exactly I should handle the cases when the size is 0, 1, or 2. Therefore, I've dealt with these situations as I find them right.

Reply:
Thanks for double-checking the logic anyway. Good to know that it's been reviewed thoroughly.